### DISCUSSION TITLE
Quansheng as modem

### DISCUSSION THREAD

**Discussion by mix127** [markqvist/Reticulum#518] [Ideas]
https://hackaday.com/2024/07/05/hacking-a-quansheng-handheld-to-transmit-digital-modes/
Quansheng UV-K6 is good idea to make a ideal modem for raspberry pi
for example USB cable <-> audio/mic 

Second options is moving some protocol to radio similar NUNU https://github.com/brett-todd/uv-k5-firmware-custom-nunu

**Comment by buhalo1** [comment]
It would be much more interesting to use the built-in MSK1200/2400 modem.
But none of the coders who invented and implemented a bunch of the most senseless bells and whistles considered it necessary to add BIIS signaling, DAPNET POCSAG transmitter and a MSK1200/2400 modem functionality to the firmware.

As for NUNU, I think it's very ridiculous to try to chat via a device not designed for this purpose. It seems more reasonable to provide the radio with the ability to receive incoming notifications, and to use an application on a smartphone to send these notifications or exchange messages in a chat.

**Comment by markqvist** [comment]
As the article mentions, one of the big problems with these kinds of radios is the very slow TX/RX turnaround time. With that out of they way though, as done by the mods mentioned in the article, you can already use it for Reticulum comms using any packet modem via the [Reticulum KISS interface](https://markqvist.github.io/Reticulum/manual/interfaces.html#kiss-interface).

Radios that are already built for this kind of thing will also allow using more efficient digital modes. Have a look at [this video where Reticulum is running over a 300 baud HF packet mode](https://youtu.be/blwNVumLujc).

I completely second that using the built-in digital modem is a much better approach, though. And the firmware could be modded to expose that over USB or serial, it would make it very easy to use these kinds of radios with Reticulum

**Reply by mix127** [reply]
All other inventions are duct tape-glued self-builds. And the use of an internal modem is just one cable that perhaps could be wired into a cell phone or raspberry pi.

The number of cables and power problems drops by half. If there is no hardware that you can just buy and use never such a network will be popular.
